File: 백준/Silver/10828. 스택/스택.swift
let N = Int(readLine()!)!
var array = [String]()

for _ in 1...N {
    let order = readLine()!.split(separator: " ").map { "\($0)" }
    switch order[0] {
    case "push":
        array.append(order[1])
    case "pop":
        guard array.isEmpty else {
            print(array.removeLast())
            continue
        }
        print("-1")
    case "size":
        print("\(array.count)")
    case "empty":
        print("\(array.isEmpty ? 1 : 0)")
        
    case "top":
        guard array.isEmpty else {
            print(array.last!)
            continue
        }
        print("-1")
    default:
        break
    }
}

File: 백준/Silver/10828. 스택/스택.py
n = int(input())
stack = []
result = ""
for _ in range(n):
    command = input().split()
    if command[0] == "push":
        stack.append(int(command[1]))
    elif command[0] == "pop":
        if not stack:
            result += "-1\n"
        else:
            result += str(stack.pop()) + "\n"
    elif command[0] == "size":
        result += str(len(stack)) + "\n"
    elif command[0] == "empty":
        result += str(1 if not stack else 0) + "\n"
    elif command[0] == "top":
        if not stack:
            result += "-1\n"
        else:
            result += str(stack[len(stack)-1]) + "\n"
    else:
        continue
print(result, end="")